WordPress/gutenberg
# Style book: create static categories

Part of:

- https://github.com/WordPress/gutenberg/issues/64707

## What? How?

MVP for https://github.com/WordPress/gutenberg/issues/53431

- Creates a static category map based on list defined in #53431 
- Sorts examples based on static categories and their subcategories
- Introduces simple layout for subcategories (under theme)


## Why?

> [!NOTE]
> This is a foundational PR only. So clearing a path to creating custom tabs and example lists. More design stuff to come in subsequent PRs. 

Below is the theme tab for example. The "Design" tab's examples are now under "Theme" sorted into subcategories.


## Testing Instructions
Open the style book and check for regressions, e.g., 

1. Click on examples. They should open the global styles panel for that block.
2. Navigate between tabs
3. Check that the stylebook can be opened from the "Style" menu item in the site editor "view" mode navigation menu. 

## Screenshots

| Trunk | This PR |
|--------|--------|
| <img width="500" alt="Screenshot 2024-09-20 at 12 52 11 pm" src="https://github.com/user-attachments/assets/61109574-a9f5-4b4d-a1b3-be6fdbad332d"> | <img width="500" alt="Screenshot 2024-09-20 at 12 51 55 pm" src="https://github.com/user-attachments/assets/768a63cb-9caa-44a9-bbc7-263921cfa269"> | 

## Steps after this PR:

- Implement layout/design in #53431 
- Create custom "Color" pattern to display theme's palettes/other colors.
- Create/collate example blocks for the landing category. Include said category. Add click events that open corresponding category tabs.
- Look at how themes can define their own examples for the "Overview" category.
- Ensure registered categories not in the static list are appended to the static list via `getCategories()` for extenders etc. https://github.com/WordPress/gutenberg/pull/65507
- Come up with a way for the UI to display lots of categories.
- Create missing block examples where appropriate.
